services/supervisor/src/lib.rs

```diff
@@ -76,6 +81,27 @@ impl Service for Supervisor {
 
             trace!("Removed outdated deployment request {:?}", request);
         }
+
+        let entry = schema.config_propose_with_hash_entry().get();
+        if let Some(entry) = entry {
+            if entry.config_propose.actual_from <= height {
+                // Remove pending config proposal for which deadline was exceeded.
+                trace!("Removed outdated config proposal");
+                schema.config_propose_with_hash_entry().remove();
+            } else {
+                // Apply pending config in case 2/3+1 validators voted for it.
+                let config_confirms = schema.config_confirms();
+                let confirmations = config_confirms.confirmations(&entry.propose_hash);
+                let validators = config_confirms.validators_len();
+
+                if confirmations >= byzantine_quorum(validators) {
+                    // Perform the application of configs.
+                    context.update_config(entry.config_propose.changes);
```

Comment:
Maybe add an `info` log entry about new config being accepted (and maybe also include the new config in it)?